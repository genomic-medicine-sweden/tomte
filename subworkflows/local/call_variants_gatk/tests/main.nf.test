nextflow_workflow {

    name "Test Workflow CALL_VARIANTS_GATK"
    script "subworkflows/local/call_variants_gatk/main.nf"
    workflow "CALL_VARIANTS_GATK"
    config "./nextflow.config"
    tag "subworkflow"
    tag "call_variants_gatk"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
        script "../../../../modules/nf-core/gunzip/main.nf"
        process {
            """
            input[0] = Channel.of([
                [ id:'fasta' ],
                file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
            ])
            """
            }
        }

        run("SAMTOOLS_FAIDX") {
        script "../../../../modules/nf-core/samtools/faidx/main.nf"
        process {
            """
            input[0] = GUNZIP_FASTA.out.gunzip
            input[1] = [[],[]]
            input[2] = []
            """
            }
        }

        run("GATK4_CREATESEQUENCEDICTIONARY", alias: "BUILD_DICT") {
        script "../../../../modules/nf-core/gatk4/createsequencedictionary/main.nf"
        process {
            """
            input[0] = GUNZIP_FASTA.out.gunzip
            """
            }
        }

    }


    test("Should run without failures") {

        when {
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.of([
                    [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                    file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                ])
                input[1] = GUNZIP_FASTA.out.gunzip.collect()
                input[2] = SAMTOOLS_FAIDX.out.fai.collect()
                input[3] = BUILD_DICT.out.dict.collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcf.collect{path(it[1]).vcf.summary},
                    file(workflow.out.tbi.get(0).get(1)).name,
                    workflow.out.stats,
                    workflow.out.versions,
                ).match()}
            )
        }

    }

}
