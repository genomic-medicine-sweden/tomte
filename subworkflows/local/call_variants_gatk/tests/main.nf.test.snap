{
    "Should run without failures": {
        "content": [
            [
                "VcfFile [chromosomes=[chr21], sampleCount=1, variantCount=2496, phased=false, phasedAutodetect=false]"
            ],
            [
                [
                    {
                        "case": "finequagga",
                        "id": "ACC5963A1",
                        "sample": "ACC5963A1",
                        "strandedness": "reverse",
                        "sex": "NA",
                        "single_end": false,
                        "fq_pairs": 1,
                        "is_fastq": false
                    },
                    "ACC5963A1_gatk.vcf.gz.tbi:md5,d7f6f8ded10dcfcac48f6fb08052985a"
                ]
            ],
            [
                [
                    {
                        "case": "finequagga",
                        "id": "ACC5963A1",
                        "sample": "ACC5963A1",
                        "strandedness": "reverse",
                        "sex": "NA",
                        "single_end": false,
                        "fq_pairs": 1,
                        "is_fastq": false
                    },
                    "ACC5963A1.bcftools_stats.txt:md5,09a5b7beb572c5754f44f2dea20024a2"
                ]
            ],
            [
                "versions.yml:md5,1e6c4168df6c725d8d8257c0983e1c77",
                "versions.yml:md5,3625fcc3c1900a0db9df0dcbba7dab0e",
                "versions.yml:md5,65d3400f4aaf64a558b891ef3f80a2c4",
                "versions.yml:md5,809a79c3a84402ca4129418a19fcf556",
                "versions.yml:md5,ca18eaf0ba5dfed37cd58ee8368be488"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T17:08:40.34165"
    }
}